ocaml/lambda/translcore.ml

```diff
@@ -985,6 +985,45 @@ and transl_exp0 ~in_new_scope ~scopes sort e =
     let l = transl_exp ~scopes sort e in
     if Config.stack_allocation then Lexclave l
     else l
+  | Texp_src_pos ->
+    let fields =
+      Datarepr.labels_of_type
```

Comment:
This sequence of steps (the `labels_of_type` followed by `Array.of_list`) will be repeated for every `Texp_src_pos`. Maybe just compute all this once right in Predef? If that causes trouble (e.g. module dependency loops) then you can put the result in a top-level value right here in Translcore.